From e2dae1732757c7c9da4dde62d1978ceea95a4e5b Mon Sep 17 00:00:00 2001
From: Liu Jian <liujian56@huawei.com>
Date: Mon, 9 Jan 2023 11:14:32 +0800
Subject: [PATCH 1373/1398] [Huawei] Revert "[Backport] tcp: fix a
 signed-integer-overflow bug in tcp_add_backlog()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188217

--------------------------------

This reverts commit 3cd03457dabcb34bae70a1608fcc6d25013e2791.

Revert this maillist inclusion patch, and backport it from LTS.
The LTS patch is:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=4f23cb2be530

Signed-off-by: Liu Jian <liujian56@huawei.com>
---
 net/ipv4/tcp_ipv4.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/ipv4/tcp_ipv4.c b/net/ipv4/tcp_ipv4.c
index 752e3256313e..3a62b63fc8fd 100644
--- a/net/ipv4/tcp_ipv4.c
+++ b/net/ipv4/tcp_ipv4.c
@@ -1772,8 +1772,7 @@ int tcp_v4_early_demux(struct sk_buff *skb)
 
 bool tcp_add_backlog(struct sock *sk, struct sk_buff *skb)
 {
-	u32 limit = (u32)READ_ONCE(sk->sk_rcvbuf) +
-		    (u32)(READ_ONCE(sk->sk_sndbuf) >> 1);
+	u32 limit = READ_ONCE(sk->sk_rcvbuf) + READ_ONCE(sk->sk_sndbuf);
 	u32 tail_gso_size, tail_gso_segs;
 	struct skb_shared_info *shinfo;
 	const struct tcphdr *th;
-- 
2.17.1

